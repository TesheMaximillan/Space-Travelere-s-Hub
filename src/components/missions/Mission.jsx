/* eslint-disable react/prop-types */
import React from 'react';
import { useDispatch } from 'react-redux';
import { updateMissionStatus } from '../../redux/missions/actions';
import Status from './Status';

function Mission({ mission }) {
  const dispatch = useDispatch();
  const {
    missionId, missionName, description, status,
  } = mission;

  const handleStatus = () => dispatch(updateMissionStatus(missionId, !status));

  return (
    <tr>
      <td>{missionName}</td>
      <td>{description}</td>
      <Status status={status} handleStatus={handleStatus} />
    </tr>
  );
}

export default Mission;
