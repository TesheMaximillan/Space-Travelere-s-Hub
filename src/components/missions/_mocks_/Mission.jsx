/* eslint-disable react/prop-types */
import React from 'react';
import { useDispatch } from 'react-redux';
import { updateMissionStatus } from '../../../redux/missions/actions';
import Status from '../Status';

function Mission() {
  const mission = {
    missionId: '1',
    missionName: 'Teshome',
    description: 'adfasdf as fads f',
    status: true,
  };
  const dispatch = useDispatch();

  const handleStatus = () => dispatch(updateMissionStatus(mission.missionId, !mission.status));

  return (
    <tr>
      <td>{mission.missionName}</td>
      <td>{mission.description}</td>
      <Status status={mission.status} handleStatus={handleStatus} />
    </tr>
  );
}

export default Mission;
