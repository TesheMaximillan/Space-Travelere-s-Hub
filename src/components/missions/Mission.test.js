import React from 'react';
import renderer from 'react-test-renderer';
import renderContext from '../storeMock';
import Mission from './_mocks_/Mission';

describe('Mission component', () => {
  let component;

  beforeEach(() => {
    component = renderer.create(renderContext(<Mission />));
  });

  it('should render with given state from redux store', () => {
    // Assert
    expect(component.toJSON()).toMatchSnapshot();
  });
});
