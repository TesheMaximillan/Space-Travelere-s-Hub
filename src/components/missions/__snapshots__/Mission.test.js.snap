// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission component should render with given state from redux store 1`] = `
<tr>
  <td>
    Teshome
  </td>
  <td>
    adfasdf as fads f
  </td>
  <td>
    <span
      className="statusActive"
    >
      Active Member
    </span>
  </td>
  <td>
    <button
      className="btnMission btnLeave"
      onClick={[Function]}
      type="button"
    >
      Leave Mission
    </button>
  </td>
</tr>
`;
